### PR TITLE
feat: RIR Tracking & Intra-Session Load Suggestions (Phase 1)

### LINKED ISSUE #2: feat: RIR Tracking & Auto-Suggestion for Load Progression
## Problem
Users have to guess how much weight to add or reps to do on the next set. No intelligent progression guidance.

## Solution
Add manual RIR (Reps in Reserve) input after each set, then use that data to suggest optimal load/rep adjustments for next set.

### Design
After user checks off a set as complete:
1. Show optional "RIR" input field (0-5 scale)
2. Save RIR to database with exercise history
3. On next set (same exercise), show suggestion:
   - If RIR was 0-1: suggest +5% load or -1 rep
   - If RIR was 3-5: suggest same load or +1 rep

### Tasks
1. Add RIR field to set completion dialog
2. Update Supabase schema to store RIR
3. Build RIR history query function
4. Implement suggestion algorithm
5. Display suggestions in set-row UI (badge-style)

## Acceptance Criteria
- [ ] RIR input appears after set validation
- [ ] RIR is saved to history
- [ ] Suggestions display for next set
- [ ] Users can accept/ignore suggestion
- [ ] RIR data tracked in analytics

## Notes
- Manual input only (no auto-detection)
- Inspiration: MyFitnessPal, Strong app

## Effort: Medium

### PR DESCRIPTION
**Pull request by PierreTsia**
## What

- Add a **RIR (Reps in Reserve) bottom sheet** that opens on every set completion — users rate effort on a 0-4+ scale via color-coded toggle buttons
- Implement an **intra-session suggestion engine** that auto-adjusts the next set's weight based on the RIR just logged (RIR 0 → decrease, RIR 1-3 → same, RIR 4+ → increase)
- Extend `set_logs` schema with a nullable `rir` column, update TypeScript types and sync service to persist RIR data
- Add comprehensive **test coverage**: 6 RirDrawer tests, 8 SetsTable tests (including suggestion edge cases), 18 syncService tests (including rir passthrough)
- Add a Cursor rule to prevent unnecessary `useEffect` + `setState` patterns

## Why

Users currently have no way to track perceived effort per set. Without effort data, load progression is pure guesswork — either too conservative (stalling progress) or too aggressive (risking injury). RIR tracking captures this signal at the moment it's most reliable (immediately after a set) and feeds it into automated weight suggestions, removing friction from the "what should I lift next?" decision.

## How

- **RIR Drawer**: Built from shadcn `Drawer` (vaul), `ToggleGroup`, `Popover`, and `Button` — zero custom UI components. Dismiss = confirm with default RIR (2) to minimize friction.
- **Two-phase set completion flow**: Checkbox tap opens the Drawer; confirm/dismiss runs the existing done logic (1RM, PR detection, rest timer, sync) with RIR injected. Uses `key` prop for state reset instead of `useEffect`.
- **Suggestion engine**: Pure function in `rirSuggestion.ts` — absolute increments (2.5 kg / 5 lbs), fully unit-tested.
- **Schema**: Single `ALTER TABLE` migration adding nullable `smallint` column. Backward-compatible with old offline queue items (`undefined` → `null`).
- **i18n**: 12 new keys per locale (EN/FR) with proper French accents.

Closes #2

### CODE REVIEW

File: supabase/migrations/20260315000001_add_rir_to_set_logs.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE set_logs ADD COLUMN rir smallint;
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply supabase/migrations/20260315000001_add_rir_to_set_logs.sql]: Addressed in `53ac6e0` (53ac6e0bc948a3896cf53f74cb53ed6619b297f4)

File: src/components/workout/RirDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const [selectedRir, setSelectedRir] = useState(DEFAULT_RIR)
+
+  function handleOpenChange(nextOpen: boolean) {
+    if (!nextOpen) {
+      onConfirm(selectedRir)
+    }
+  }
+
+  function handleConfirm() {
+    onConfirm(selectedRir)
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/workout/RirDrawer.tsx]: Intentional design decision: dismiss = confirm with default RIR is specified in the Tech Plan to minimize mobile friction. Adding a cancel path would require an extra tap on every set completion, which is the UX we specifically designed against.

File: src/lib/rirSuggestion.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export function computeIntraSessionSuggestion(
+  prevRir: number,
+  prevWeight: number,
+  prevReps: string,
+  unit: WeightUnit,
+): IntraSessionSuggestion {
+  const inc = INCREMENT[unit]
+
+  if (prevRir === 0) {
+    return { weight: Math.max(inc, prevWeight - inc), reps: prevReps }
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/lib/rirSuggestion.ts]: Addressed in `53ac6e0` (53ac6e0bc948a3896cf53f74cb53ed6619b297f4)

File: src/locales/fr/workout.json [hunk 4]
```diff
@@ -39,5 +39,17 @@
   "skip": "Passer",
   "restOverNotif": "Repos terminé !",
   "restOverBody": "C'est parti pour la prochaine série",
-  "finishEarly": "Terminer la séance"
+  "finishEarly": "Terminer la séance",
+  "rir.title": "Répétitions en réserve",
+  "rir.description": "Combien de répétitions aurais-tu encore pu faire ?",
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/locales/fr/workout.json]: Addressed in `53ac6e0` (53ac6e0bc948a3896cf53f74cb53ed6619b297f4)
